File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun sumOfLeftLeaves(root: TreeNode?): Int {
        if (root === null) return 0
        var sum = 0
        if (root.left !== null && root.left.left === null && root.left.right === null) {
            sum += root.left.`val`
        }
        sum += sumOfLeftLeaves(root.left)
        sum += sumOfLeftLeaves(root.right)
        return sum
    }
}

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        stack, res = [], 0
        while stack or root:
            if root:
                stack.append(root)
                root = root.left
                if root and not root.left and not root.right:
                    res += root.val
            else:
                root = stack.pop()
                root = root.right
        return res
        
        #
        
        if not root:
            return 0
        res = 0
        res += root.left.val if root.left and not root.left.left and not root.left.right else 0
        return res + self.sumOfLeftLeaves(root.left) + self.sumOfLeftLeaves(root.right)
            